classes/Module/AddonsCurlClient.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Module;
+
+use PrestaShop\Module\AutoUpgrade\Tools14;
+
+/**
+ * Class AddonsCurlClient is a simple Addons client that uses Curl to perform its request.
+ */
+class AddonsCurlClient implements AddonsClientInterface
```

Comment:
Is this class a copy from the Core?

Reply:
Not an exact copy, but the code comes from the core yes, but I prefer to have a dedicated class to manage this call inside the module. Since we don't know on which version the module is installed we can't rely on the core classes because they are modified from a version to another.
I think that's why the `Tools` class was copied in `Tools14`

Reply:
By the way I think I copied this code from the `Tools` class but I can't remember which version (since I was working on different versions at the moment). I also removed a few switch cases because the API simply didn't work with these parameters.

Reply:
I was only asking because of snake case and some parts about `services` and `hosted_module` :sweat_smile: